app/Http/Controllers/CartController.php

```diff
@@ -95,4 +100,111 @@ public function delete(Request $request, $id)
         return redirect()->back()
             ->with('success', __('Remove successful'));
     }
+
+    public function checkout()
+    {
+        if (!session()->has('cart')) {
+            return redirect()->route('cart')
+                ->with('error', __('Please add items to your cart'));
+        }
+
+        return view('user.checkout');
+    }
+
+    public function order(Request $request)
+    {
+        if (!session()->has('cart')) {
+            return redirect()->route('cart')
+                ->with('error', __('Please add items to your cart'));
+        }
+
+        $cart = $request->session()->get('cart');
+
+        $shipping_info = $request->validate([
+            'name' => 'required|string|min:5|max:255',
+            'address' => 'required|min:10|max:255',
+            'phone' => 'required|digits:10|min:10',
+            'note' => 'nullable|string',
+        ]);
+
+        // Begin transaction
+        DB::beginTransaction();
```

Comment:
tại sao chỗ này em lại dùng transaction?

Reply:
Khi thực hiện nhiều thao tác thay đổi dữ liệu đồng thời mà các thao tác có liên quan đến nhau, nếu một thao tác bị lỗi khiến cho dữ liệu không còn nguyên vẹn.
Vì vậy cần tạo transaction để đảm bảo tất cả các thao tác dữ liệu đều được thực thành công, nếu lỗi thì toàn bộ thao tác sẽ bị hủy bỏ về trạng thái trước khi xảy ra transaction tránh cập nhập nửa vời.